Source/EnhancedBeliefs/GameComponent_EnhancedBeliefs.cs

```diff
@@ -149,11 +149,20 @@ public static int BeliefDifferences(Ideo ideo1, Ideo ideo2)
 
         public void BaseOpinionRecache(Ideo ideo)
         {
+            if (ideo == null) return; // optionally Log.Message something?
             List<Pawn> pawns = pawnTrackerData.Keys.ToList();
 
             for (int i = 0; i < pawns.Count; i++)
             {
-                pawnTrackerData[pawns[i]].baseIdeoOpinions[ideo] = pawnTrackerData[pawns[i]].DefaultIdeoOpinion(ideo);
+                if (pawns[i].Ideo == ideo)
+                {
+                    pawnTrackerData.Remove(pawns[i]);
+                    AddTracker(pawns[i]); // reset tracker completely since their ideo changed
```

Comment:
...What is this even for? Removing a tracker deletes all ideology-related memories, it doesn't make sense that a pawn would suddenly stop hating vegans after being indoctrinated in dozens of debates and conversions just because they decided to worship insects all of a sudden

Reply:
Ah! I see - didn't understand the details of how the IdeoTrackers work. What's the appropriate way to force a rebalancing of the opinions?
```C#
if (pawns[i].Ideo == ideo)
{
  foreach (var i in pawnTrackerData[pawns[i]].baseIdeoOpinions.Keys)
    pawnTrackerData[pawns[i]].baseIdeoOpinion[i] = pawnTrackerData[pawns[i]].DefaultIdeoOpinion(i);
}

Reply:
What is the point of clearing tracker data anyways? The whole point of RecacheBaseOpinions is to do this exact thing to all ideos. Base opinion of pawn's own ideo always returns their certainty, not the baseIdeoOpinion value.